service/storage_service.cc

```diff
@@ -145,6 +145,10 @@ namespace {
     return string_list_to_endpoint_list(utils::split_comma_separated_list(comma_separated_list));
 }
 
+[[nodiscard]] bool validate_ignore_nodes_params(const std::vector<sstring>& src_node_strings) {
+    return std::ranges::all_of(src_node_strings, [](const auto& node_str) { return locator::host_id_or_endpoint{node_str}.has_endpoint(); });
```

Comment:
So if all are IP addresses, we return `true`.

Below we warn if the opposite is true (i.e. one of them is a host ID).

This means you will print the warning if a host ID appears, and not print it if every entry is an IP address. This is exactly the opposite of what we want to do...

You need to test the change at least manually.

Reply:
Isn't `validate_host_ids` that you added to scylla-nodetool.cc exactly the same function you want to sure here?

Reply:
My bad, did not recheck after this change.

Reply:
Fixed in the new version